tests/unit/changeset-test.js

```diff
@@ -685,6 +682,60 @@ module('Unit | Utility | changeset', function(hooks) {
     assert.equal(c.get('foo'), get(model, 'foo'));
   });
 
+  test('#set works after save', async function(assert) {
+    dummyModel['org'] = {
+      usa: {
+        mn: 'mn',
+        ny: 'ny'
+      }
+    };
+
+    const c = new Changeset(dummyModel);
+    c.set('org.usa.ny', 'NY');
+    c.set('org.usa.mn', 'MN');
+
+    assert.equal(c.get('org.usa.ny'), 'NY');
+    assert.equal(c.get('org.usa.mn'), 'MN');
+    assert.equal(dummyModel.org.usa.ny, 'ny');
+    assert.equal(dummyModel.org.usa.mn, 'mn');
+
+    c.save();
+
+    assert.equal(c.get('org.usa.ny'), 'NY');
+    assert.equal(c.get('org.usa.mn'), 'MN');
+    assert.equal(dummyModel.org.usa.ny, 'NY');
+    assert.equal(dummyModel.org.usa.mn, 'MN');
+
+    c.set('org.usa.ny', 'nil');
+
+    assert.equal(c.get('org.usa.ny'), 'nil');
+    assert.equal(c.get('org.usa.mn'), 'MN');
+    assert.equal(dummyModel.org.usa.ny, 'NY');
+    assert.equal(dummyModel.org.usa.mn ,'MN');
+
+    c.save();
+
+    assert.equal(c.get('org.usa.ny'), 'nil');
+    assert.equal(c.get('org.usa.mn'), 'MN');
+    assert.equal(dummyModel.org.usa.ny, 'nil');
+    assert.equal(dummyModel.org.usa.mn, 'MN');
+
+    c.set('org.usa.ny', 'nil2');
+    c.set('org.usa.mn', 'undefined');
+
+    assert.equal(c.get('org.usa.ny'), 'nil2');
+    assert.equal(c.get('org.usa.mn'), 'undefined');
+    assert.equal(dummyModel.org.usa.ny, 'nil');
+    assert.equal(dummyModel.org.usa.mn, 'MN');
+
+    c.save();
+
+    assert.equal(c.get('org.usa.ny'), 'nil2');
+    assert.equal(c.get('org.usa.mn'), 'undefined');
+    assert.equal(dummyModel.org.usa.ny, 'nil2');
+    assert.equal(dummyModel.org.usa.mn, 'undefined');
+  });
```

Comment:
@josemarluedke Does this test look right to you?

Reply:
Yep. Looks good! Interesting that it is passing. Not sure how to capture that bug in a test then.